### PR TITLE
Update item

### PR DESCRIPTION
**Pull request by Matt-Kou**

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -320,82 +329,61 @@ def test_get_item(self):
...
         order = self._create_orders(1)[0]
         item = OrderItemFactory()
-        response = self.app.post(
+        
+        # add item
+        resp = self.app.post(
             f"{BASE_URL}/{order.id}/items",
             json=item.serialize(),
             content_type="application/json",
         )
```
Comment (MahamArif) [inline tests/test_routes.py:342]: Why you changed an existing test in your PR?

File: service/routes.py [hunk 2]
```diff
@@ -211,6 +211,41 @@ def get_item(order_id, item_id):
...
 
+######################################################################
+# UPDATE AN ITEM FROM AN ORDER
+######################################################################
+@app.route("/orders/<int:order_id>/items/<int:item_id>", methods=["PUT"])
+def update_item(order_id, item_id):
+    """
+    Update an item from an order
+    """
+    app.logger.info("Request to read an Item %s from Order with id: %s", item_id, order_id)
```
Comment (MahamArif) [inline service/routes.py:222]: It should be request to update order, not read!

File: service/routes.py [hunk 3]
```diff
@@ -211,6 +211,41 @@ def get_item(order_id, item_id):
...
+    item = OrderItem.find_by_order_and_item_id(order_id, item_id)
+    if not item:
+        abort(
+            status.HTTP_404_NOT_FOUND,
+            f"Item with id '{item_id}' could not be found with order id '{order_id}'.",
+        )
+        
+    data = request.get_json()
+    item.deserialize(data)
+    # TODO: remove
```
Comment (MahamArif) [inline service/routes.py:242]: Please remove TODO

File: tests/test_routes.py [hunk 4]
```diff
@@ -426,3 +414,58 @@ def test_get_item_order_not_found(self):
         """
         resp = self.app.get(f"{BASE_URL}/0/items/0")
         self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_update_no_item(self):
+        """It should not Update an item given wrong order id and non-existent order"""
```
Comment (MahamArif) [inline tests/test_routes.py:419]: Please create separate test cases to check different corner cases.

File: tests/test_routes.py [hunk 5]
```diff
@@ -426,3 +414,58 @@ def test_get_item_order_not_found(self):
...
+        # update non-existent item id
+        updated_item = OrderItemFactory()
+        updated_item.id = item.id + 123
+        resp = self.app.put(
+            f"{BASE_URL}/{order.id}/items/{updated_item.id}",
+            json=updated_item.serialize(),
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
```
Comment (MahamArif) [inline tests/test_routes.py:447]: You should also check here what is not found, order or item. Check the message of the response like other test cases!

**Review by MahamArif — APPROVED** [review]
